pkg/plugin/settings.go

```diff
@@ -48,14 +50,38 @@ func loadSettings(appSettings backend.AppInstanceSettings) Settings {
 	}
 
 	switch settings.OpenAI.Provider {
-	case openAIProviderOpenAI:
-	case openAIProviderAzure:
+	case OpenAIProviderOpenAI:
+	case OpenAIProviderAzure:
 	default:
 		// Default to OpenAI if an unknown provider was specified.
 		log.DefaultLogger.Warn("Unknown OpenAI provider", "provider", settings.OpenAI.Provider)
-		settings.OpenAI.Provider = openAIProviderOpenAI
+		settings.OpenAI.Provider = OpenAIProviderOpenAI
 	}
 
 	settings.OpenAI.apiKey = appSettings.DecryptedSecureJSONData[openAIKey]
-	return settings
+
+	// embedder settings
+	// If the embedding setup option is set to OpenAI, then we need to use openAIKey and URL
+	// Otherwise we use the auth setting and URL from the vector store.
+	var authType embed.EmbedderAuthType
+	var URL string
+	embedderType := embed.EmbedderType(settings.Vector.Embed.Type)
```

Comment:
Could we use the same case with default fallback here to avoid weird errors? (e.g. fallback to OpenAI if not provided)